pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -534,9 +534,18 @@ public int getPGType(String pgTypeName) throws SQLException {
       ResultSet rs = castNonNull(oidStatement.getResultSet());
       if (rs.next()) {
         oid = (int) rs.getLong(1);
-        String internalName = castNonNull(rs.getString(2));
-        oidToPgName.put(oid, internalName);
-        pgNameToOid.put(internalName, oid);
+        // when this function is called it will be called with a fully qualified type name including the schema
+        // if the schema part is public then we can get rid of the schema, otherwise keep it.
+        // there is an outstanding question of how this works if the search path has changed
+        if (pgTypeName.toLowerCase(Locale.ROOT).startsWith("public.")) {
```

Comment:
What is the reason for an explicit `startsWith` check here?
Could you add tests that exercise it?

If you still want to compare it, I would suggest `String.regionMatches(ignoreCase=true, ...`, so it does not need to convert the thing to lowercase on every invocation.